src/types.cpp

```diff
@@ -369,7 +369,11 @@ SizedType CreateBuffer(size_t size)
 
 SizedType CreateTimestamp()
 {
-  return SizedType(Type::timestamp, 64);
+  auto tp = SizedType(Type::timestamp, 24);
+  tp.tuple_elems.emplace_back(Type::integer, 8);
+  tp.tuple_elems.emplace_back(Type::integer, 8);
+  tp.tuple_elems.emplace_back(Type::integer, 8);
```

Comment:
I don't think using `tuple_elems` is correct. Why is this necessary?

Reply:
I made it this way to avoid an error "invalid indirect read from stack off -40+8 size 16". I'll look for another solution I guess.